src/modes/ModeEstimation.jl

```diff
@@ -147,6 +147,32 @@ function (f::OptimLogDensity)(z)
     return -DynamicPPL.getlogp(varinfo)
 end
 
+function (f::OptimLogDensity)(F, G, z)
+    spl = DynamicPPL.SampleFromPrior()
+    
+    # Calculate log joint and the gradient
+    l, g = gradient_logp(
+        z, 
+        DynamicPPL.VarInfo(f.vi, spl, z), 
+        f.model, 
+        DynamicPPL.SampleFromPrior(),
+        f.context
+    )
+
+    if G !== nothing
```

Comment:
If `G === nothing` it seems wasteful to compute the gradient at all. I guess in that case we should only evalute the log joint probability.

Reply:
Fixed.